### PR TITLE
HIP-584: Create integration tests for testing dynamic eth_call using create and update token precompiles

### PR DESCRIPTION
**Pull request by bilyana-gospodinova**
**Description**:
We need to create integration tests that validate eth_call works correctly in dynamic cases using create and update precompiles to validate that they update the in-memory state properly.

Fixes the following bugs:
- SyntheticTxnFactory#createTokenCreate does not add the keys.
- AssociatePrecompile creates TokenRelationship with frozen = true and kycGranted = false (from this PR: https://github.com/hashgraph/hedera-mirror-node/pull/6613).
- FunctionEncodeDecoder#encodeToken does not map the keys.

**This PR covers the following scenarios**:
- Update token key + get token info key
- Update token expiry + get token info expiry
- Update token info that updates symbol + get token info symbol
- Update token info that updates name + get token info name
- Update token info that updates memo + get token info memo
- Update auto renew period + get token info auto renew period
- Delete fungible/non-fungible token + get token info isDeleted
- CreateToken for fungible token with/without default freeze/kyc status + name + symbol + decimals + getDefaultFreezeStatus + getDefaultKycStatus + isToken
- CreateToken for nft with/without default freeze/kyc status + name + symbol + tokenUri + ownerOf + getDefaultFreezeStatus + getDefaultKycStatus + isToken

Fixes https://github.com/hashgraph/hedera-mirror-node/issues/6585

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java [hunk 1]
```diff
@@ -96,12 +96,23 @@ public TokenRelationship(Token token, Account account, boolean hasAssociation) {
                 token.isFrozenByDefault() && token.hasFreezeKey(),
                 !token.hasKycKey(),
                 false,
-                true,
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java:99]: I believe the default value for `notYetPersisted` should be `true`. If this causes a problem, the issue might be elsewhere and fixed at the appropriate place.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java [hunk 2]
```diff
@@ -96,12 +96,23 @@ public TokenRelationship(Token token, Account account, boolean hasAssociation) {
...
 
+    /**
+     * Constructor for creating new TokenRelationship with AssociatePrecompile
+     * @param token
+     * @param account
+     * @param hasAssociation
+     * @param frozen
+     */
+    public TokenRelationship(Token token, Account account, boolean hasAssociation, boolean frozen) {
+        this(token, account, 0, frozen, true, false, false, false, hasAssociation, 0);
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java:113]: Here we assume `kycGranted` is `true`. I'm not sure that would be always correct. Probably we can also pass this value as a parameter.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java (hunk 1 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java]: Changed to true. There are no failing tests after the change.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java (hunk 2 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/models/TokenRelationship.java]: kycGranted is now passed as a parameter. The value is synced from services.

File: hedera-mirror-web3/src/test/resources/contracts/NestedEthCallsTestContract/NestedEthCallsTestContract.sol [hunk 3]
```diff
@@ -0,0 +1,175 @@
+pragma solidity ^0.8.0;
+
+import "./HederaTokenService.sol";
+import "./HederaResponseCodes.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract NestedEthCalls is HederaTokenService {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/resources/contracts/NestedEthCallsTestContract/NestedEthCallsTestContract.sol:7]: We could make this smart contract agnostic of what type of calls are executed over it - `eth_call` or `eth_estimateGas` and this contract naming could be made more universal. So we can rename it to `NestedCalls` only.

Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/resources/contracts/NestedEthCallsTestContract/NestedEthCallsTestContract.sol]: The contract is now renamed.

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM

**Review by 0xivanov — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**